File: Bellini/DataAccessLayer/Models/Quiz.cs
namespace DataAccessLayer.Models
{
    public class Quiz : BaseModel
    {
        public string GameName { get; set; } = null!;
        public string GameCoverImageUrl { get; set; } = null!;
        public bool IsDraft { get; set; } = false;
        public DateTime StartTime { get; set; } = DateTime.MinValue;
        public DateTime EndTime { get; set; } = DateTime.MinValue;
        public List<QuizQuestion> Questions { get; set; } = null!;
        public List<QuizResults> QuizResults { get; set; } = null!;
        public List<QuizComment> Comments { get; set; } = null!;
    }
}

File: Bellini/client.react/src/utils/interfaces/Quiz.ts
import {QuizQuestion} from "@/utils/interfaces/QuizQuestion.ts";

export interface Quiz {
    id: number;
    gameName: string;
    gameCoverImageUrl: string;
    questions: QuizQuestion[];
    hasUserCompleted: boolean;
    quizResults: QuizResult[];
}

export interface QuizResult {
    numberOfCorrectAnswers: number;
    numberOfQuestions: number;
    userId: number;
    isReplay: boolean;
    endTime: string;
}